src/lib/prover/prover.ml

```diff
@@ -66,6 +66,7 @@ module Worker_state = struct
     ; logger : Logger.Stable.Latest.t
     ; proof_level : Genesis_constants.Proof_level.t
     ; constraint_constants : Genesis_constants.Constraint_constants.t
+    ; commit_id : string
```

Comment:
This seems pretty bad. If we have a prover process running on a different version that the main mina binary, it will claim to be the same version as the Mina binary. I strongly suggest reintroducing `Mina_version` as a dependency here.

Reply:
I assumed it's impossible for `prover` and `verifier` to run a different executable from their master process (given they're launched via `rpc_parallel`)